internal/services/tableland_api_service.go

```diff
@@ -0,0 +1,62 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/DIMO-Network/identity-api/internal/config"
+	"github.com/rs/zerolog"
+	"net/http"
+	"net/url"
+)
+
+type TablelandApiService struct {
+	log      *zerolog.Logger
+	settings *config.Settings
+}
+
+func NewTablelandApiService(log *zerolog.Logger, settings *config.Settings) *TablelandApiService {
+	return &TablelandApiService{
+		log:      log,
+		settings: settings,
+	}
+}
+
+func (r *TablelandApiService) Query(ctx context.Context, queryParams map[string]string, result interface{}) error {
+	fullURL, err := url.Parse(r.settings.TablelandAPIGateway)
+	if err != nil {
+		return err
+	}
+
+	fullURL = fullURL.JoinPath(fullURL.Path, "api/v1/query")
+
+	if queryParams != nil {
+		values := fullURL.Query()
+		for key, value := range queryParams {
+			values.Set(key, value)
+		}
+		fullURL.RawQuery = values.Encode()
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fullURL.String(), nil)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Perhaps we should use a client with a timeout, or use a context with some deadline.